src/DbConnectedResource.php

```diff
@@ -1,30 +1,51 @@
 <?php
 /**
  * @license   http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
- * @copyright Copyright (c) 2014 Zend Technologies USA Inc. (http://www.zend.com)
+ * @copyright Copyright (c) 2014-2016 Zend Technologies USA Inc. (http://www.zend.com)
  */
 
 namespace ZF\Apigility;
 
 use Zend\Db\TableGateway\TableGatewayInterface as TableGateway;
 use Zend\Paginator\Adapter\DbTableGateway as TableGatewayPaginator;
+use ZF\ApiProblem\Exception\DomainException;
 use ZF\Rest\AbstractResourceListener;
 
 class DbConnectedResource extends AbstractResourceListener
 {
+    /**
+     * @var string Name of collection class
+     */
     protected $collectionClass;
 
+    /**
+     * @var string Name of identifier field
+     */
     protected $identifierName;
 
+    /**
+     * @var TableGateway
```

Comment:
Ooof, that's a hard call. Changing that would be a BC break, as it's tightening the type hint (vs loosening); simultaneously, you are correct in the assertion that the `DbTableGateway` adapter requires the more specific type.

What we should likely do is update the `DbTableGateway` to accept `TableGatewayInterface` instead (it predates that interface, which is why it typehints against the concrete instance).

I'm going to leave this as-is for now.